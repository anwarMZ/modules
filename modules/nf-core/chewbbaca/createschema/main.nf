process CHEWBBACA_CREATESCHEMA {
    tag "$meta.id"
    label 'process_low'

    conda "bioconda::chewbbaca=3.3.1"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/chewbbaca:3.3.1--pyhdfd78af_0':
        'biocontainers/chewbbaca:3.3.1--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(fasta, stageAs: "input_genomes/*")
    path prodigal_tf
    path cds

    output:
    tuple val(meta), path("results/$meta.id")         , emit: schema
    path "results/cds_coordinates.tsv"                 , emit: cds_coordinates
    path "results/invalid_cds.txt"                     , emit: invalid_cds
    path "versions.yml"                                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def schema = "--n ${prefix}"
    def prodigal_tf = prodigal_tf ? "--ptf ${prodigal_tf}" : ""
    def cds = cds ? "--cds ${cds}" : ""

    """
    find ./input_genomes/ -name "*.gz" | sed 's/.gz//' | xargs -I {} bash -c 'gzip -cdf {}.gz > {}'

    chewie \\
        CreateSchema \\
        -i input_genomes/ \\
        -o results \\
        $schema \\
        $args \\
        $prodigal_tf \\
        $cds \\
        --cpu $task.cpus

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        chewbbaca: \$(echo \$(chewie --version 2>&1 | sed 's/^.*chewBBACA version: //g; s/Using.*\$//' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def schema = "--n ${prefix}"

    """
    mkdir -p results/$schema

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        chewbbaca: \$(echo \$(chewie --version 2>&1 | sed 's/^.*chewBBACA version: //g; s/Using.*\$//' ))
    END_VERSIONS
    """
}
