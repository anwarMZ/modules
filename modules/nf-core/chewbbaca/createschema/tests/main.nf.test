nextflow_process {

    name "Test Process CHEWBBACA_CREATESCHEMA"
    script "../main.nf"
    process "CHEWBBACA_CREATESCHEMA"

    tag "modules"
    tag "modules_nfcore"
    tag "chewbbaca"
    tag "chewbbaca/createschema"

    test("test_chewbbaca_createschema_single") {

        when {
            process {

                """

                ptf     = []
                cds     = []
                input[0] = [[ id:'test', single_end:false ], // meta map
                file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)]
                input[1] = ptf
                input[2] = cds
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_chewbbaca_createschema_multi") {

        when {
            process {

                """

                ptf     = []
                cds     = []
                input[0] = = [[ id:'test', single_end:false ], // meta map
                file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true),
                file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)]
                input[1] = ptf
                input[2] = cds
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_chewbbaca_createschema_gz") {

        when {
            process {

                """

                ptf     = []
                cds     = []
                input[0] = [[ id:'test', single_end:false ], // meta map
                file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true),
                file(params.test_data['sarscov2']['genome']['genome_fasta_gz'], checkIfExists: true)]
                input[1] = ptf
                input[2] = cds
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
