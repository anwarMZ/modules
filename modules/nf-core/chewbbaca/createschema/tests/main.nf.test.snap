{
    "sarscov2 illumina contigs [fasta]": {
        "content": [
            [
                ".schema_config",
                "contigs-protein1.fasta",
                "contigs-protein10.fasta",
                "contigs-protein11.fasta",
                "contigs-protein13.fasta",
                "contigs-protein14.fasta",
                "contigs-protein16.fasta",
                "contigs-protein17.fasta",
                "contigs-protein18.fasta",
                "contigs-protein19.fasta",
                "contigs-protein2.fasta",
                "contigs-protein20.fasta",
                "contigs-protein4.fasta",
                "contigs-protein6.fasta",
                "contigs-protein7.fasta",
                "contigs-protein8.fasta",
                "contigs-protein9.fasta",
                "short"
            ],
            [
                "cds_coordinates.tsv:md5,7fef49207aa024b68f465e986f4c7444"
            ],
            [
                "invalid_cds.txt:md5,d0afbd9996054d8838e62dc3e852fa07"
            ],
            [
                "versions.yml:md5,cc97e7dbcf8455a4bf166f0bf326e769"
            ]
        ],
        "timestamp": "2024-01-26T12:13:29.558920352"
    },
    "sarscov2 illumina contigs [fasta] & genome [fasta_gz]": {
        "content": [
            [
                ".schema_config",
                "contigs-protein1.fasta",
                "contigs-protein10.fasta",
                "contigs-protein11.fasta",
                "contigs-protein13.fasta",
                "contigs-protein14.fasta",
                "contigs-protein16.fasta",
                "contigs-protein17.fasta",
                "contigs-protein18.fasta",
                "contigs-protein19.fasta",
                "contigs-protein2.fasta",
                "contigs-protein20.fasta",
                "contigs-protein4.fasta",
                "contigs-protein6.fasta",
                "contigs-protein7.fasta",
                "contigs-protein8.fasta",
                "contigs-protein9.fasta",
                "short"
            ],
            [
                "cds_coordinates.tsv:md5,7fef49207aa024b68f465e986f4c7444"
            ],
            [
                "invalid_cds.txt:md5,d0afbd9996054d8838e62dc3e852fa07"
            ],
            [
                "versions.yml:md5,cc97e7dbcf8455a4bf166f0bf326e769"
            ]
        ],
        "timestamp": "2024-01-26T12:14:01.738666237"
    },
    "sarscov2 illumina contigs [fasta] & genome [fasta_gz] - stub": {
        "content": [
            [
                ".schema_config",
                "contigs-protein{1..*}.fasta",
                "short"
            ],
            [
                "cds_coordinates.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            [
                "invalid_cds.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            [
                "versions.yml:md5,cc97e7dbcf8455a4bf166f0bf326e769"
            ]
        ],
        "timestamp": "2024-01-26T12:14:12.757130114"
    },
    "sarscov2 illumina contigs & genome [fasta]": {
        "content": [
            [
                ".schema_config",
                "contigs-protein1.fasta",
                "contigs-protein10.fasta",
                "contigs-protein11.fasta",
                "contigs-protein13.fasta",
                "contigs-protein14.fasta",
                "contigs-protein16.fasta",
                "contigs-protein17.fasta",
                "contigs-protein18.fasta",
                "contigs-protein19.fasta",
                "contigs-protein2.fasta",
                "contigs-protein20.fasta",
                "contigs-protein4.fasta",
                "contigs-protein6.fasta",
                "contigs-protein7.fasta",
                "contigs-protein8.fasta",
                "contigs-protein9.fasta",
                "short"
            ],
            [
                "cds_coordinates.tsv:md5,7fef49207aa024b68f465e986f4c7444"
            ],
            [
                "invalid_cds.txt:md5,d0afbd9996054d8838e62dc3e852fa07"
            ],
            [
                "versions.yml:md5,cc97e7dbcf8455a4bf166f0bf326e769"
            ]
        ],
        "timestamp": "2024-01-26T12:13:45.580057569"
    }
}